client/rpcserver/types.go

```diff
@@ -270,3 +280,22 @@ func parseTradeArgs(params *RawParams) (*tradeForm, error) {
 	}
 	return req, nil
 }
+
+func parseCancelArgs(params *RawParams) (*cancelForm, error) {
+	if err := checkNArgs(params, []int{1}, []int{1}); err != nil {
+		return nil, err
+	}
+	id := params.Args[0]
+	if len(id) != orderIdLen {
+		return nil, fmt.Errorf("%w: orderID has incorrect length", errArgs)
+	}
+	// We can only check up to 64 bits at a time. Break the 256 bit hex
+	// into 4 sections and check each.
+	div := orderIdLen / 4
+	for i := 0; i < 4; i++ {
+		if _, err := checkUIntArg(id[i*div:(i+1)*div], "orderID", 16, 64); err != nil {
+			return nil, fmt.Errorf("%w: invalid order id hex", errArgs)
+		}
+	}
```

Comment:
I'd be way more comfortable with `_, err := hex.DecodeString(id)` and then just replicate error message formatting here.

Reply:
Note that `dex/order.IDFromHex` does both, although it has an unnecessary `copy` for our case.